app/src/main/java/info/nightscout/android/medtronic/service/MedtronicCnlIntentService.java

```diff
@@ -232,12 +270,36 @@ protected void onHandleIntent(Intent intent) {
                     cnlReader.updatePumpStatus(pumpRecord);
 
                     cnlReader.endEHSMSession();
+                    MainActivity.dbgCNL_beginEHSMSession -= 1;
 
                     if (pumpRecord.getSgv() != 0) {
+
+                        String sgvString;
+                        if (MainActivity.mmolxl) {
+                            sgvString = MainActivity.sgvFormatter.format((float) pumpRecord.getSgv() / MainActivity.MMOLXLFACTOR);
```

Comment:
Ok - I moved the conversion to MainActivity and changed back public to private for the variables used. I still needed to have them static though as the conversion function is static - is that ok?